docs/circuits.md

```diff
@@ -362,3 +362,109 @@ print(circuit[1:3])
 Especially useful is dropping the last moment (which are often just
 measurements): ``circuit[:-1]``, or reversing a circuit:
 ``circuit[::-1]``.
+
+
+### Importing from OpenQASM
+
+The QASM importer is in experimental state and currently only supports a subset of the full OpenQASM spec. 
+Amongst others, classical control, arbitrary gate definitions and even some of the gates that don't have a one-to-one representation in Cirq are not yet supported.
+The functionality should be sufficient to import interesting quantum circuits. 
+Error handling is very simple - on any lexical or syntactical error a `QasmException` is raised. 
+
+#### Importing cirq.Circuit from QASM format
+
+
+Requirements: `ply` 
+
+```bash
+pip install ply==3.4
+```
+
+The following call will create a circuit defined by the input QASM string:
+
+```python
+from cirq.contrib.qasm_import.qasm import QasmCircuitParser
+circuit = QasmCircuitParser().parse("""
```

Comment:
Why `QasmCircuitParser` defined like this? Is there anything else it would be doing other than parsing a circuit? If not, having a `parse` method defined under it is redundant. Instead of `pass` on the `init`, how about we 

```python
def __init__(self, qasm: str) -> circuits.Circuit:
       return QasmParser(qasm).circuit
```
So that this is 

```python
circuit = QasmCircuitParser("""
    OPENQASM 2.0;
    include "qelib1.inc";
    qreg q[3];
    creg meas[3];
    h q;
    measure q -> meas;
    """)
```

Reply:
@vtomole If you want a single call to give you a circuit, a better way would be to define a global function (or static method on Circuit) that does this.

Reply:
Sure. Once `qasm_import` is moved out of contrib since we don't want Circuit to depend on anything in contrib.

Reply:
I had somewhat similar design originally - but @Strilanc suggested to move out any logic from constructors (see this [thread](https://github.com/quantumlib/Cirq/pull/1613#discussion_r287132860)). 
We could have a single function on the contrib package too: 
`cirq.contrib.qasm_import.qasm.circuit_from_qasm("""OPENQASM 2.0;""")`

Reply:
I would prefer the single function *instead* of the class because the class isn't adding anything.

If you decide to keep the class, delete the empty `__init__` method.  It's not doing anything.

Reply:
I like your function suggestion. This is not really relevant to this PR, so i will open an issue for it or comment on the QASM import issue.